### PR TITLE
examples standalone in repo

### PR DESCRIPTION
**Pull request by dk1844**
The examples have been moved to `examples/` subfolder (core examples & s3 sdk extension examples). Each has their own maven pom definition and a readme.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -51,3 +51,9 @@ jobs:
       - name: Build and unit test
         if: ${{ ! contains( github.event.pull_request.labels.*.name, 'NoTestNeeded') }}
         run: mvn clean install --no-transfer-progress -Pspark-${{ matrix.spark }} -Dspan.scale.factor=10
+      - name: Build and unit test Atum Examples Scala ${{ matrix.scala }} and Spark ${{ matrix.spark }}
+        working-directory: ./examples/atum-examples
+        if: ${{ ! contains( github.event.pull_request.labels.*.name, 'NoTestNeeded') }}
+        run: |
+          mvn scala-cross-build:change-version -Pscala-${{ matrix.scala }}
+          mvn clean install --no-transfer-progress -Pspark-${{ matrix.spark }} -Dspan.scale.factor=10
```
Comment (Zejnilovic) [inline .github/workflows/build.yml:59]: Why not create a second action? `build_examples.yml`

File: examples/atum-examples/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Atum Spark Job Application Example
+
+This is a set of Atum Apache Spark Applications that can be used as inspiration for creating other
+Spark projects. It includes all dependencies in a 'fat' jar to run the job locally and on a     cluster.
```
Comment (Zejnilovic) [inline examples/atum-examples/README.md:4]:
```suggestion
Spark projects. It includes all dependencies in a 'fat' jar to run the job locally and on a cluster.
```

File: examples/atum-examples/pom.xml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!-- scala cross build -->
+            <plugin>
+                <groupId>com.github.wajda</groupId>
+                <artifactId>scala-cross-build-maven-plugin</artifactId>
+                <version>0.2.1</version>
```
Comment (Zejnilovic) [inline examples/atum-examples/pom.xml:177]: I know this is in another pom as well. But could this be moved up to properties (even in the other one)

File: examples/atum-examples/src/test/scala/za/co/absa/atum/utils/SparkJobRunnerMethods.scala [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018 ABSA Group Limited
+ * Copyright 2018-2019 ABSA Group Limited
```
Comment (Zejnilovic) [inline examples/atum-examples/src/test/scala/za/co/absa/atum/utils/SparkJobRunnerMethods.scala:2]: AFAIK we agreed on a single year here

File: examples/s3-sdk-extension-examples/src/test/scala/za/co/absa/atum/utils/SparkJobRunnerMethods.scala [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+package za.co.absa.atum.utils
+
+import org.scalatest.funsuite.AnyFunSuiteLike
+import scala.language.reflectiveCalls
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe
+
+trait SparkJobRunnerMethods {
```
Comment (Zejnilovic) [inline examples/s3-sdk-extension-examples/src/test/scala/za/co/absa/atum/utils/SparkJobRunnerMethods.scala:24]: I see this is in both examples and s3-examples. 1to1 copy. Is that needed?

**Review by Zejnilovic — COMMENTED** [review]
I like it. i found only small things, more questions

File: examples/atum-examples/src/test/scala/za/co/absa/atum/utils/SparkJobRunnerMethods.scala (hunk 4 above)
Comment (dk1844) [reply examples/atum-examples/src/test/scala/za/co/absa/atum/utils/SparkJobRunnerMethods.scala]: Fixed, thanks for noticing.

File: examples/atum-examples/pom.xml (hunk 3 above)
Comment (dk1844) [reply examples/atum-examples/pom.xml]: Thanks for noticing, fixed. While doing this, I have also moved the version of `maven-release-plugin` to properties.

File: examples/s3-sdk-extension-examples/src/test/scala/za/co/absa/atum/utils/SparkJobRunnerMethods.scala (hunk 5 above)
Comment (dk1844) [reply examples/s3-sdk-extension-examples/src/test/scala/za/co/absa/atum/utils/SparkJobRunnerMethods.scala:24]: Yes, it is 1:1 copy. Well, both projects need it to run tests and they are not depending on each other. What alternative do you have in mind?

File: .github/workflows/build.yml (hunk 1 above)
Comment (dk1844) [reply .github/workflows/build.yml:59]: I think this has multiple advantages:
 - the setup matrix is already there (the examples are thus built for Scala 2.11 + Spark2.4, Scala 2.12 + Spark 2.4, and Scala 2.12 + Spark 3.1.
 - the core Atum is already built and installed, a second workflow would have to do it all over.

So a step made more sense to me, but I am by no means an actions expert, so feel free to prove me wrong.

Comment (Zejnilovic) [reply .github/workflows/build.yml:59]: I forgot you need cached core

**Review by Zejnilovic — APPROVED** [review]

**Review by AB006J9 — APPROVED** [review]

**Review by Zejnilovic — APPROVED** [review]

**Pull request merged**